### PR TITLE
Avoided calling build_side_list_from_node_list() in THM

### LINKED ISSUE #26200: THM inadvertently alters internal side sets
## Bug Description
Some internal side sets, such as those within various regions of a 2D heat structure, get extra sides added because `boundary_info.build_node_list_from_side_list()` gets called and then later `boundary_info.build_side_list_from_node_list()` gets called.

For example, consider a 1D mesh with 2 elements, 0 and 1. If we add a sideset that contains (elem=0,side=1), the right side of element 0, pointing toward element 1, then a nodeset gets created from this sideset, and then the original sideset gets (elem=1,side=0) added.

## Steps to Reproduce
See above.

## Impact
This can cause some very baffling issues. For instance, evaluating an integral over an internal side then includes both sides, leading to a completely incorrect integral.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #26200

### CODE REVIEW

**Review by GiudGiud — APPROVED** [review]
this is awesome. I ran into big issues with FileMeshComponents because of this I had to comment it out

File: modules/thermal_hydraulics/include/components/Component1D.h [hunk 1]
```diff
@@ -25,15 +25,28 @@ class Component1D : public GeneratedMeshComponent
   {
     /// Physical position of the connecting point
     Point _position;
+    /// Boundary element
+    const Elem * _elem;
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/include/components/Component1D.h:29]: could be const Elem * const or not?

File: modules/thermal_hydraulics/include/components/Component1D.h [hunk 2]
```diff
@@ -25,15 +25,28 @@ class Component1D : public GeneratedMeshComponent
...
+    /// Boundary side
+    unsigned short int _side;
     /// Boundary node of connection (used by other components for connecting)
-    Node * _node;
+    const Node * _node;
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/include/components/Component1D.h:33]: same question

**Comment by GiudGiud** [comment]
This broke some vtb input.

What do you prefer for the fix?
- add an option to still run it
- finish component as actions and naturally be able to add a Mesh block with the boolean to create the sides there

**Comment by joshuahansel** [comment]
It should be a simple code fix along the same lines as this one. I'll put up a PR soon.

**Comment by joshuahansel** [comment]
https://github.com/idaholab/moose/pull/26302

**Pull request merged**